main.tf

```diff
@@ -118,6 +119,6 @@ module "dns" {
   aliases          = "${var.aliases}"
   parent_zone_id   = "${var.parent_zone_id}"
   parent_zone_name = "${var.parent_zone_name}"
-  target_dns_name  = "${aws_cloudfront_distribution.default.domain_name}"
-  target_zone_id   = "${aws_cloudfront_distribution.default.hosted_zone_id}"
+  target_dns_name  = "${element(concat(aws_cloudfront_distribution.default.*.domain_name, list("")), 0)}"
```

Comment:
More readable syntax:

```
"${join("", aws_cloudfront_distribution.default.*.domain_name)}"
```

we use the `splat+join` pattern everywhere.
`join` will return an empty string if the list does not contain any items.